### PR TITLE
Add user delete operation to users API documentation and postman collection

### PR DESCRIPTION
**Pull request by brianzuzga-okta**
@mystiberry-okta @georgekwon-okta Please review.

FYI: @keithchin-okta @daytonwilliams-okta @donwoods-okta 

Resolves: OKTA-94138

### CODE REVIEW

**Comment by brianzuzga-okta** [comment]
This feature is currently in internal beta.  It will go EA at some unspecified time in November.  I'm not sure if there are additional, conditional tags that I need to add to control the visibility of this new API call

**Comment by mystiberry-okta** [comment]
Hi @brianzuzga-okta ! Here's the link for how to add Beta tags if you're adding Beta content to a document that also has published content. https://oktawiki.atlassian.net/wiki/display/eng/Okta+Developer+Site+Authoring

If you need help with it, I can help around Thur. or Fri.

File: _source/_docs/api/resources/users.md [hunk 1]
```diff
@@ -2267,7 +2267,7 @@ Content-Type: application/json
 
 <span class="api-uri-template api-uri-post"><span class="api-label">POST</span> /users/*:id*/lifecycle/unsuspend</span>
 
-Unsuspends users and returns then to the `ACTIVE` state.  This operation can only be performed on users that have a `SUSPENDED` status.
+Unsuspends a user and returns them to the `ACTIVE` state.  This operation can only be performed on users that have a `SUSPENDED` status.
```
Comment (donwoods-okta) [inline _source/_docs/api/resources/users.md]: Is the trailing whitespace harmless or significant?

File: _source/_docs/api/resources/users.md [hunk 2]
```diff
@@ -2304,6 +2304,50 @@ HTTP/1.1 200 OK
...
+--------- | ------------ | ---------- | -------- | -------- | -------
+id        | `id` of user | URL        | String   | TRUE     |
+
+##### Response Parameters
+{:.api .api-response .api-response-params}
+
+Returns an empty object. 
+
+Passing an invalid `id` returns a `404 Not Found` status code with error code `E0000007`.
+Passing an `id` that is not in the `DEPROVISIONED` state returns a `400 Bad Request` status code with error code `E0000001`.
```
Comment (donwoods-okta) [inline _source/_docs/api/resources/users.md]: Do we normally include what happens if the `id` is for a user for whom the current session is not an admin?

Comment (mystiberry-okta) [inline _source/_docs/api/resources/users.md]: I've never seen a special note like that.

File: _source/_docs/api/resources/users.md (hunk 1 above)
Comment (mystiberry-okta) [inline _source/_docs/api/resources/users.md]: Harmless.

File: _source/_docs/api/resources/users.md [hunk 3]
```diff
@@ -2304,6 +2304,50 @@ HTTP/1.1 200 OK
 Content-Type: application/json
 ~~~
 
+### Delete User
+{:.api .api-operation}
+
+<span class="api-uri-template api-uri-post"><span class="api-label">POST</span> /users/*:id*/lifecycle/delete</span>
```
Comment (karlmcguinness-okta) [inline _source/_docs/api/resources/users.md]: You are going to get a lot of questions on why this operation is not a HTTP delete.  Can we provide an explanation to help folks understand?

**Comment by mystiberry-okta** [comment]
@brianzuzga-okta  Will you add the info, or do you want to explain to me and I can add the content Karl requests.

File: _source/_docs/api/resources/users.md (hunk 3 above)
Comment (brianzuzga-okta) [inline _source/_docs/api/resources/users.md]: @karlmcguinness-okta As I'm reading the docs more closely, it appears we've never documented the fact that DELETE on the User entity causes deactivation.  Should I also add that?  It makes our design a little more understandable.  I wanted to make sure this wasn't intentionally omitted.  (Unlikely, but I thought I'd check.)

**Comment by mystiberry-okta** [comment]
@brianzuzga-okta just a ping--are more changes coming for this PR?

**Comment by brianzuzga-okta** [comment]
Many more changes are coming based off Karl's feedback.

**Comment by mystiberry-okta** [comment]
Thanks for the update!

**Comment by brianzuzga-okta** [comment]
@mystiberry-okta I believe I have the doc correct now.  I'm just going to test my revised postman collection more.

**Comment by brianzuzga-okta** [comment]
Ok.  I'm happy with my postman collection.

**Comment by mystiberry-okta** [comment]
@brianzuzga-okta We moved some content around in users.md, do you want me to resolve the conflict (since I caused it)? We just switched the order of model and endpoint sections.

**Comment by mystiberry-okta** [comment]
@brianzuzga-okta Actually, can you resolve the conflict (we just flopped the order of data model and endpoint sections)? I'm bouncing between branches and a little worried I'll make a mistake with your branch. In an ordinary week I could fix this.

File: _source/_docs/api/resources/users.md [hunk 4]
```diff
@@ -2323,6 +2323,55 @@ HTTP/1.1 200 OK
...
+
+### Delete User
+{:.api .api-operation}
+
+<span class="api-uri-template api-uri-post"><span class="api-label">DELETE</span> /users/*:id*</span>
+
+Deletes a user permanently.  This operation can only be performed on users that have a `DEPROVISIONED` status.  **This action cannot be recovered!**
+
+This operation on a user entity that has not been deactivated will cause that user to be deactivated.  A second DELETE
+would be required to affect the full deletion.
```
Comment (donwoods-okta) [inline _source/_docs/api/resources/users.md]: affect --> effect

Comment (brianzuzga-okta) [inline _source/_docs/api/resources/users.md]: Wow.  There's a corner of affect vs effect that I had wrong all these years.

**Comment by brianzuzga-okta** [comment]
Fixed.  How does it look now?

**Comment by mystiberry-okta** [comment]
Merge for tomorrow's site refresh?

**Comment by brianzuzga-okta** [comment]
Works for me.  I hear people are going to be excited by finally having delete.

**Pull request merged**